torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -426,7 +427,13 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
                          << " encountered error when writing outgoing request: "
                          << error.what();
           }
-          markFutureWithError(std::move(futureResponseMessage), error.what());
+          auto it = clientPipe.pendingResponseMessage_.find(requestMessageId);
+          TORCH_INTERNAL_ASSERT(
+              it != clientPipe.pendingResponseMessage_.end(),
```

Comment:
I don't know whether we should assert this, as it could in fact reasonably happen: if the pipeRead callback for the response to a previous request receives an error then it's going to mark all pending futures with that error and clear the map, which means that this code will not be able to find the future. But, in that case, it should just keep going rather than raising an exception.

Reply:
I see, yeah I just saw the error handling code where we flush all messages belonging to a particular pipe in case of an error in the pipeRead callback.

Reply:
Updated to remove this.